CHK-316: Initial Setup

wdsrocha opened the pull request:
#### Description

The setup is based on [this document](https://docs.google.com/document/d/11DewV9j4qiz_U_tuLsAHUa235oydbcSupIkuVdMi9kg), https://github.com/vtex-apps/graphql-interface-app-example and https://github.com/vtex-apps/search-graphql (which is also an interface).

The schema is based on a subset of https://github.com/vtex-apps/places-graphql/ schemas (see https://github.com/vtex-apps/places-graphql/pull/5). After CHK-340 is complete, this subset of schemas on Places GraphQL will be removed.

#### How should this be manually tested?

Just check if the files are correct and make sense, there is not too much to test besides that. I suggest reviewing per commit.

wdsrocha committed:
Add CHANGELOG

wdsrocha committed:
Add Github config files

wdsrocha committed:
Add dotfiles

wdsrocha committed:
Add GraphQL types and schema

Address types are duplicated from `place-graphql`. How to avoid this?

wdsrocha committed:
Update README

wdsrocha commented on graphql/schema.graphql:
```diff
@@ -0,0 +1,5 @@
+type Query {
+  suggestAddresses(searchTerm: String): [AddressSuggestion]
+
+  getAddressByExternalId(id: String): Address
```
Is there a way to import the `Address` type from the https://www.github.com/vtex-apps/places-graphql ?

lucasecdb replied:
I don't think so :-/

lucasecdb commented on .gitignore:
```diff
@@ -0,0 +1,2 @@
+.vscode/
+.DS_Store
```
isn't it missing more stuff? like node_modules etc

wdsrocha replied:
I felt the same way, but as there is no `package.json`, there isn't much to add. On the example project there is an extensive `.gitgnore`, but looks exaggerated and unnecessary to me. What do you say?

the file: https://github.com/vtex-apps/graphql-interface-app-example/blob/c78a9a1a2a4f197dccfd302eb36088beb97a4eb7/graphql-interface-app/.gitignore

lucasecdb replied:
I'd rather add it now because we might want something like prettier to auto format the graphql queries

lucasecdb commented on docs/README.md:
````diff
@@ -0,0 +1,13 @@
+To use it in your app, declare it on your manifest file like:
+
+```
+"dependencies": {
+  "vtex.geolocation-graphql-interface": "0.x"
+}
````
this could be just
```suggestion
vtex add vtex.geolocation-graphql-interface
```

wdsrocha committed:
Update gitignore

Now it's equal to the base example https://github.com/vtex-apps/graphql-interface-app-example/blob/c78a9a1a2a4f197dccfd302eb36088beb97a4eb7/graphql-interface-app/.gitignore

wdsrocha committed:
Add editorconfig

wdsrocha committed:
Add yarn.lock

wdsrocha committed:
Release v0.0.2-0
